### PR TITLE
Colormap_diff Fix

### LINKED ISSUE #541: Diff. Custom Colourmap Broken
**Describe the bug**
When you change the diff colourmap in a comparison, it doesn't do anything

**To Reproduce**
Do a diff. then change the colourmap

**Expected behavior**
The area plot should reload with the new selected colourmap

**Screenshots**

image

**Additional context**
Erica Discovered this problem

### PR DESCRIPTION
**Pull request by NoahGallant-MUN**
Resolves #541 

Recently, the colormap when drawing an area map for dataset comparison was broken.

This fixes it.

Fixed Plot:
image

### CODE REVIEW

File: oceannavigator/frontend/src/components/AreaWindow.jsx [hunk 1]
```diff
@@ -98,33 +98,42 @@ export default class AreaWindow extends React.Component {
...
+            output_starttime: this.props.dataset_0.time,
+            output_endtime: this.props.dataset_0.time
           }
         );
       }
     }
   }
 
+  /*componentWillReceiveProps(props) {
+    if (this._mounted && stringify(this.props) !== stringify(props)) {
```
Comment (htmlboss) [inline oceannavigator/frontend/src/components/AreaWindow.jsx]: If we're not using this code anymore, might as well delete it. If we need it in the future we can look back in the git history

File: plotting/map.py [hunk 2]
```diff
@@ -344,6 +344,7 @@ def load_data(self):
 
         if self.compare:
             self.variable_name += " Difference"
+            self.cmap = cmap = colormap.find_colormap(self.compare.get('colormap_diff'))
```
Comment (htmlboss) [inline plotting/map.py]: I looked further down the file and I can see why you've got the chained assignment thing going on here. What you've is fine for now.

**Comment by htmlboss** [comment]
Another cool thing I learned about github PRs a while ago is that if you put "Resolves ISSUE_NUMER" in the initial PR message, github will automatically close the linked issue when the PR is merged. I've made an edit to your message as an example. Also, when you navigate to that issue, you'll see this PR linked to it at the bottom of the timeline.

File: plotting/map.py (hunk 2 above)
Comment (NoahGallant-MUN) [reply plotting/map.py]: I did not even know you could do that, or that I had done it.
Should I i get rid of the cmap assignment? lol

File: oceannavigator/frontend/src/components/AreaWindow.jsx (hunk 1 above)
Comment (NoahGallant-MUN) [reply oceannavigator/frontend/src/components/AreaWindow.jsx]: Will do

**Comment by NoahGallant-MUN** [comment]
Do we want to do that to attach the pull request? I've been using the 'connect this pull request with an existing issue' option that zenhub gives and it moves it around it's panels.

File: oceannavigator/frontend/src/components/AreaWindow.jsx (hunk 1 above)
Comment (NoahGallant-MUN) [reply oceannavigator/frontend/src/components/AreaWindow.jsx]: > Judging from the change from `componentWillReceiveProps` to `componentDidUpdate`, React wasn't updating the state properly?
> 
> Anyway other than the comment about the unneeded code, it's good to go. Would it also be possible to attach a screenshot of the fixed plot in the original post? I find it helps with constructing a timeline in the event something breaks down the line.

The intention was to fix the reloading issue. It didn't work.
That being said, componentWillReceiveProps throws deprecation warnings, so we should switch over eventually anyway.

File: plotting/map.py (hunk 2 above)
Comment (htmlboss) [reply plotting/map.py]: LOL well I think it's needed since  `cmap` (not `self.cmap`) is referenced below here.

**Pull request merged**